atsign-foundation/at_server
# fix: refactor compaction job

**- What I did**
* Consume the https://github.com/atsign-foundation/at_server/pull/1113 PR changes and implement the refactor of the compaction job.
* Add generics to the AtCommitLog, AtAccessLog and AtNotificationLog classes for type safety.
* Remove the time-based compaction and size-based compaction files because that are not in use.
* Add unit tests for the compaction.
* Updated at_commons and at_persistence_spec versions in pubspec.yaml

**- How to verify it**
* Unit tests to verify the changes to commit log compaction
* The tests in the at_secondary_server should pass to ensure that there are no breaking changes.

**- Description for the changelog**
* fix: Refactor commit log compaction to enable compaction on the client side.
